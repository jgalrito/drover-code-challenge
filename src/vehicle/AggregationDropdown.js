import React from 'react'

import {capitalize} from '../utils/string'

import Dropdown from '../common/components/Dropdown'

const AggregationDropdown = ({counts, options, onChange}) => {
  let aggregationOptions = Object.keys(counts || {})

  if(options.transformOptions)
    aggregationOptions = options.transformOptions(aggregationOptions)

  const getLabel = key => (options.translations && options.translations[key]) || capitalize(key)

  return (
    <div className="input-group-no-conflict">
      <label>{typeof options === 'string'?options:options.label}</label>
      <Dropdown
        options={[undefined, ...aggregationOptions]}
        valueLabel={key => key?getLabel(key):'Any'}
        optionLabel={key => key?`${getLabel(key)} (${counts[key]})`:'Any'}
        onChange={onChange}
      />
    </div>
  ) 
}

export default AggregationDropdown