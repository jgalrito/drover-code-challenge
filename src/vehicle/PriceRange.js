import React, {Component} from 'react'
import {Range} from 'rc-slider'
import styled from 'styled-components'

import 'rc-slider/assets/index.css'

const PriceRangeWrapper = styled.div`
	.range-container {
		padding: 0 6px;

		.rc-slider-track, .rc-slider-handle {
			border: 2px solid #172B24;
    		background-color: #50ff7d;
		}
	}
`

class PriceRange extends Component {
	state = {
		value: this.props.value
	}

	onChange = value => this.setState({value})

	render = () => {
		const
			{value} = this.state,
			[min, max] = value

		return (
		  <PriceRangeWrapper className="price-range">
		  	<p>£{min} - £{max}</p>
		  	<div className="range-container">
		    	<Range min={100} max={2500} step={10} value={value} onChange={this.onChange} onAfterChange={() => this.props.onChange(value)}/>
		    </div>
		  </PriceRangeWrapper>
		)	
	}
}

export default PriceRange