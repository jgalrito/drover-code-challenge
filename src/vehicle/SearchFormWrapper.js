import styled from 'styled-components'

const SearchFormWrapper = styled.div`
  .search-form .input-group-no-conflict > * {
    display: block;
  }

  .sort {
    display: flex;
    justify-content: flex-end;
    align-items: center;

    .label {
      font-weight: bold;
      display: flex;
      align-items: center;
      border-right: 1px solid #e4e3e3;

      img {
        height: 25px;

        &.asc {
          transform: scale(1, -1);
        }
      }
    }

    .dropdown {
      .dropdown-value {
        height: 30px;
        padding: 0;
        border: none;
        background-color: transparent;
      }
    }
  }

  .search-results > h3 {
    text-transform: uppercase;
  }

  .subscription-type {
    border-radius: 6px;
    background: #ffffff;
    border: 1px solid #e4e4e4;

    .subscription-type-picker {
      .label {
        font-weight: bold;
      }

      .subscription-type-option {
        display: flex;
        align-items: center;

        label {
          font-weight: normal;
          font-size: 14px;
        }
      }
    }

    .subcription-duration-picker {
      display: flex;
      align-items: center;
      justify-content: flex-end;

      .dropdown ul {
        max-height: 301px;
      }
    }
  }

  .result-list {
    list-style: none;
    
    & > :not(:last-child) {
      margin-bottom: 15px;
    }
  }

  .pagination-wrapper {
    text-align: center;
    color: #172B24;

    .pagination {
      justify-content: center;

      li {
        &:not(:last-child) {
          margin-right: 10px;
        }

        &.prev, &.next {
          a {
            background-color: #bef9f9;
          }
        }

        &.active a {
          background-color: #50ff7d;
        }

        a {
          border: none;
          padding: 10px;
          text-align: center;
          border-radius: 6px;
          background: #ffffff;
          text-decoration: none;
          color: inherit;
        }
      }
    }
  }
`

export default SearchFormWrapper