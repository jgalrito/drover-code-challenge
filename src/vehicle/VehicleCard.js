import React from 'react'
import moment from 'moment'

import {capitalize} from '../utils/string'

import Button from '../common/components/Button'

import VehicleCardWrapper from './VehicleCardWrapper'
import VehicleSlideShow from './VehicleSlideShow'
import discountTag from './assets/discount_tag.png'

import {VEHICLE_MAKES, SUB_TYPES} from './'

const PCO_KEY_FACTS = {
  sub_type: {
    icon: require('./assets/uber.svg'),
    label: key => SUB_TYPES[key]
  },
  city_jurisdiction: {
    icon: require('./assets/city_jurisdiction.svg')
  }
}

const KEY_FACTS = {
  year: {
    icon: require('./assets/year.svg')
  },
  body_information: {
    icon: require('./assets/body_type.svg')
  },
  transmission: {
    icon: require('./assets/transmission.svg')
  },
  fuel: {
    icon: require('./assets/fuel_type.svg')
  },
  number_seats_information: {
    icon: require('./assets/seats.svg'),
    label: seats => `${seats} Seats`
  },
  number_doors_information: {
    icon: require('./assets/doors.svg'),
    label: doors => `${doors} Doors`
  },
  color: {
    icon: require('./assets/color.svg')
  }
}

const FEATURES = {
  air_conditioning: "Air Conditioning",
  aircon: "Air Conditioning",
  alloy_wheels: "Alloy Wheels",
  aux_input: "Aux Input",
  bluetooth: "Bluetooth Connectivity",
  convertible: "Convertible",
  cruise_control: "Cruise Control",
  dab_radio: "DAB Radio",
  electric_sliding_doors: "Electric Sliding Doors",
  electric_windows: "Electric Windows",
  fmam_radio: "FM/AM Radio",
  heated_seats: "Heated Seats",
  leather_seats: "Leather Seats",
  parking_sensors: "Parking Sensors",
  satellite_navigation: "Satellite Navigation",
  satnav: "Satellite Navigation",
  start_stop: "Auto Start-Stop",
  tinted_windows: "Tinted Windows",
  upholstery_cloth: "Upholstery Cloth"
}

export const VehicleCard = ({vehicle, isConsumer, totalPrice, discount}) => {
  const {
    images,
    vehicle_make,
    vehicle_model,
    engine_size_information,
    postcode,
    available_start_date,
    features,
    year,
    fuel
  } = vehicle,
    vehicleMake = VEHICLE_MAKES[vehicle_make.toLowerCase()] || vehicle_make,
    engineSize = engine_size_information && fuel !== 'electric'?` ${engine_size_information}L`:'',
    showYear = isConsumer?'':` - ${year}`,
    title = `${vehicleMake} ${vehicle_model}${showYear}${engineSize}`

  const generateKeyFacts = keyFacts => Object.entries(keyFacts).map(([key, {icon, label}]) => {
    if(!vehicle[key])
      return null

    label = capitalize((label?label(vehicle[key]):vehicle[key]).toString())

    return (
      <li key={key} className="px-1 pb-1">
        <img src={icon} alt={label}/>
        <span>{label}</span>
      </li>
    ) 
  })

  return (
    <VehicleCardWrapper className="vehicle-card">
      <div className="slide-show-container">
        <VehicleSlideShow images={images}/>
        {discount > 0 && (
          <img alt="discount" src={discountTag}/>
        )}
      </div>
      <div className="vehicle-info p-1 p-lg-0">
        <div className="vehicle-title pl-3">
          <div className="pb-1 pt-2">
            <h3 className="vehicle-make-brand m-0">{capitalize(title)}</h3>
            <span className="vehicle-location">Located in {postcode.split(' ')[0]}</span>
          </div>
          <div className="availability pr-3">
            Available from {moment(available_start_date).format('Do MMMM YYYY')}
          </div>
        </div>
        <div className="key-facts py-1 px-2">
          <ul>
            {!isConsumer && generateKeyFacts(PCO_KEY_FACTS)}
            {generateKeyFacts(KEY_FACTS)}
          </ul>
        </div>
        <ul className="features py-1 px-2">
          {features.reduce((items, feature, i) => {
            items.push(FEATURES[feature])

            if(i < features.length - 1)
              items.push('•');

            return items;
          }, []).map((content, i) => <li key={i} className="pl-1 pr-1">{content}</li>)}
        </ul>
        <div className="pricing px-2 pb-2 pt-0 p-lg-2 p-xl-2 px-xl-3">
          {discount > 0 && (
            <div className="discount-info">
              <div><small className="max-price">was £{totalPrice + discount}/month</small></div>
              <div><small>(save £{discount}/month)</small></div>
            </div>
          )}
          <div className="value-wrapper mr-lg-3 p-2 p-lg-0 mb-2 mb-lg-0">
            {discount > 0 && <small className="max-price">£{totalPrice + discount}</small>}
            <div className="value">
              <span className="price text-xl-right">£ {totalPrice}</span>/month
            </div>
            <small className="periodicity">
              (Monthly Vehicle Price inc. VAT)
            </small>
          </div>
          <Button>See more details</Button>
        </div>
      </div>
    </VehicleCardWrapper>
  )
}

export default VehicleCard