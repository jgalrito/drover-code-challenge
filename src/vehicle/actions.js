import {search, SUBSCRIPTION_START_OPTIONS} from './'

export const
  CLEAR_RESULTS = 'CLEAR_RESULTS',
  FETCH_VEHICLES_START = 'FETCH_VEHICLES_START',
  FETCH_VEHICLES_SUCCESS = 'FETCH_VEHICLES_SUCCESS',
  UPDATE_SEARCH_PARAMS = 'UPDATE_SEARCH_PARAMS'

const clearResults = () => ({type: CLEAR_RESULTS})

export const fetchVehicles = () => async (dispatch, getState) => {
  dispatch({type: FETCH_VEHICLES_START})

  //param names that don't match between API input and response
  const {body_information, sortBy, ...searchParams} = getState().vehicle.searchParams

  dispatch({
    type: FETCH_VEHICLES_SUCCESS,
    results: await search({
      ...searchParams,
      ...sortBy,
      body_type: body_information
    })
  })
}

export const setVehicleType = vehicle_type => dispatch => {
  const subscriptionStartOptions = SUBSCRIPTION_START_OPTIONS[vehicle_type]

  //clean results since consumer type vehicles don't have PCO pricing information
  dispatch(clearResults())

  dispatch(updateSearchParams({
    subscription_start_days: subscriptionStartOptions[subscriptionStartOptions.length - 1],
    vehicle_type
  }))
}

export const updateSearchParams = update => async (dispatch, getState) => {
  dispatch({
    type: UPDATE_SEARCH_PARAMS,
    update
  })

  dispatch(fetchVehicles())
}