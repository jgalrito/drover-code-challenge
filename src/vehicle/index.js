export const search = ({
  city_jurisdiction,
  body_type,
  fuel,
  location,
  max_distance,
  number_of_months,
  number_of_weeks,
  order_by,
  order_direction,
  page,
  price_min,
  price_max,
  rolling,
  vehicle_type,
  sub_type,
  subscription_start_days,
  tags,
  transmission,
  vehicle_make,
  year
}) => fetch('https://app.joindrover.com/api/web/vehicles', {
  method: 'POST',
  headers: {
    'Content-Type': 'application/json'
  },
  body: JSON.stringify({
    city_jurisdiction,
    body_type,
    fuel,
    location,
    max_distance,
    number_of_months,
    number_of_weeks,
    order_by,
    order_direction,
    page,
    per_page: 15,
    price_min,
    price_max,
    rolling,
    vehicle_type,
    sub_type,
    subscription_start_days,
    tags: tags && [tags],
    transmission,
    vehicle_make,
    year: parseInt(year, 10) || undefined
  })
}).then(res => res.json())

export const VEHICLE_MAKES = {
  bmw: 'BMW',
  byd: 'BYD',
  mini: 'MINI'
}

export const SUB_TYPES = {
  uber_exec: 'Uber Exec',
  uber_lux: 'Uber Lux',
  uberx: 'Uber X',
  uberxl: 'Uber XL'
}

export const SUBSCRIPTION_START_OPTIONS = {
  consumer: [2, 14, 30],
  PCO: [2, 7]
}

export const SORTING_OPTIONS = [{
  order_by: 'price',
  order_direction: 'asc',
  label: 'Price - Low To High'
}, {
  order_by: 'price',
  order_direction: 'desc',
  label: 'Price - High To Low'
}, {
  order_by: 'distance',
  order_direction: 'asc',
  label: 'Distance - Close To Far'
}]

export const DISTANCE_OPTIONS = [25, 50, 75, 100, 150, 200]