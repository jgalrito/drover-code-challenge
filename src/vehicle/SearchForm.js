import React, { Component } from 'react'
import Pagination from 'react-js-pagination'
import {connect} from 'react-redux'

import sortIcon from '../assets/icons/sort.svg'

import Location from '../common/components/Location'
import Dropdown from '../common/components/Dropdown'
import Radio from '../common/components/Radio'
import Button from '../common/components/Button'
import CaretIcon from '../common/components/CaretIcon'
import LoadingSpinnerIcon from '../common/components/LoadingSpinner'

import VehicleCard from './VehicleCard'
import PriceRange from './PriceRange'
import AggregationDropdown from './AggregationDropdown'
import SearchFormWrapper from './SearchFormWrapper'

import {updateSearchParams, setVehicleType, fetchVehicles} from './actions'
import {VEHICLE_MAKES, SUB_TYPES, SUBSCRIPTION_START_OPTIONS, SORTING_OPTIONS, DISTANCE_OPTIONS} from './'

const PCO_AGGREGATION_TYPES = {
  city_jurisdiction: 'City Jurisdiction',
  sub_type: {
    label: 'Uber Type',
    transformOptions: keys => keys.filter(key => key !== 'not_approved'),
    translations: SUB_TYPES
  },
}

const AGGREGATION_TYPES = {
  vehicle_make: {
    label: 'Vehicle Make',
    translations: VEHICLE_MAKES
  },
  transmission: 'Gearbox',
  year: {
    label: 'Year',
    transformOptions: years => years.reverse()
  },
  fuel: 'Fuel Type',
  tags: 'Car Type',
  body_information: 'Body Type'
}

const VEHICLE_TYPES = {
  consumer: 'Consumer',
  PCO: 'Private-Hire'
}

export class SearchForm extends Component {
  state = {
    refineSearch: false
  }

  componentDidMount = () => this.props.fetchVehicles()

  toggleRefineSearch = () => this.setState({
    refineSearch: !this.state.refineSearch
  })

  render = () => {
    const {
      refineSearch
    } = this.state, {
      loading,
      results,
      searchParams,
      updateSearchParams,
      setVehicleType
    } = this.props, {
      location,
      number_of_months,
      number_of_weeks,
      sortBy,
      price_min,
      price_max,
      vehicle_type,
      subscription_start_days,
      max_distance,
      rolling
    } = searchParams, {
      aggregations,
      page,
      total_count,
      per_page
    } = ((results && results.metadata) || {}),
      totalPages = total_count && Math.floor(total_count / per_page),
      isConsumer = vehicle_type === 'consumer'

    const generateAggregationOptions = aggregationOptions => Object.entries(aggregationOptions).map(([key, options]) => (
      <AggregationDropdown
        key={key}
        counts={aggregations && aggregations[key]}
        options={options}
        value={searchParams[key]}
        onChange={value => updateSearchParams({[key]: value})}
      />
    ))

    return (
      <SearchFormWrapper className={refineSearch?'refine-search':''}>
        <div className="back-to-results p-3" onClick={this.toggleRefineSearch}>
          <CaretIcon stroke="white"/>
          <span className="ml-2">Back to your search results</span>
        </div>
        <div className="container py-3">
          <div className="row">
            <div className="search-form col-12 col-lg-3">
              <div className="input-group-no-conflict">
                <label>Vehicle Type</label>
                <Dropdown
                  options={Object.keys(VEHICLE_TYPES)}
                  valueLabel={value => VEHICLE_TYPES[value]}
                  optionLabel={value => VEHICLE_TYPES[value]}
                  value={vehicle_type}
                  onChange={setVehicleType}
                />
              </div>
              <div className="input-group-no-conflict">
                <label>Location</label>
                <Location onChange={location => updateSearchParams({location})} value={location}/>
              </div>
              <div className="input-group-no-conflict">
                <label>Subscription starts within the</label>
                <Dropdown
                  options={SUBSCRIPTION_START_OPTIONS[vehicle_type]}
                  valueLabel={value => `${value} Days`}
                  optionLabel={value => `Next ${value} Days`}
                  value={subscription_start_days}
                  onChange={subscription_start_days => updateSearchParams({subscription_start_days})}
                />
              </div>
              <div className="input-group-no-conflict">
                <label>Distance (radius in miles)</label>
                <Dropdown
                  options={DISTANCE_OPTIONS}
                  value={max_distance}
                  onChange={subscription_start_days => updateSearchParams({subscription_start_days})}
                />
              </div>
              <div className="input-group-no-conflict">
                <label>Monthly Budget</label>
                <PriceRange
                  value={[price_min, price_max]}
                  onChange={([price_min, price_max]) => updateSearchParams({price_min, price_max})}
                />
              </div>
              {!isConsumer && generateAggregationOptions(PCO_AGGREGATION_TYPES)}
              {generateAggregationOptions(AGGREGATION_TYPES)}
            </div>
            <div className="search-results col-12 col-lg-9">
              {!loading && results && <h3>{total_count} vehicles found near {location}</h3>}
              {loading && <h3>Searching for vehicles near {location}</h3>}
              {results && (
                <div className="subscription-type">
                  <div className="subscription-type-picker px-3 pt-2 pb-1">
                    <div className="label my-1">Choose your subscription type:</div>
                    {isConsumer && (
                      <div className="consumer-subscription-options">
                        <div className="subscription-type-option mb-2" onClick={() => updateSearchParams({rolling: true})}>
                          <Radio id="rolling-true" checked={!!rolling} readOnly/>
                          <label className="m-0 pl-2" htmlFor="rolling-true">Monthly Rolling Subscription (cancel or swap monthly)</label>
                        </div>
                        <div className="subscription-type-option" onClick={() => updateSearchParams({rolling: false})}>
                          <Radio id="rolling-false" checked={!rolling} readOnly/>
                          <label className="m-0 pl-2" htmlFor="rolling-false">Minimum Commitment Subscription (get discounts!)</label>
                        </div>
                      </div>
                    )}
                    {!isConsumer && (
                      <div className="pco-subscription-options">
                        <div className="subscription-type-option mb-2" onClick={() => updateSearchParams({rolling: true})}>
                          <Radio id="rolling-true" checked={!!rolling} readOnly/>
                          <label className="m-0 pl-2" htmlFor="rolling-true">Weekly Rolling Subscription (cancel or swap weekly)</label>
                        </div>
                        <div className="subscription-type-option" onClick={() => updateSearchParams({rolling: false})}>
                          <Radio id="rolling-false" checked={!rolling} readOnly/>
                          <label className="m-0 pl-2" htmlFor="rolling-false">2 to 52 weeks (Subscribe for longer and save!)</label>
                        </div>
                      </div>
                    )}
                  </div>
                  {!rolling && (
                    <div className="subcription-duration-picker pr-3 pl-3 pb-2">
                      Subscription duration
                      {isConsumer && (
                        <div className="consumer-subscription-duration">
                          <Dropdown
                            className="mx-2"
                            value={number_of_months}
                            options={Array(11).fill().map((_, i) => i+2)}
                            onChange={number_of_months => updateSearchParams({number_of_months})}
                          />
                          months
                        </div>
                      )}
                      {!isConsumer && (
                        <div className="pco-subscription-duration">
                          <Dropdown
                            className="mx-2"
                            value={number_of_weeks}
                            options={Array(51).fill().map((_, i) => i+2)}
                            onChange={number_of_weeks => updateSearchParams({number_of_weeks})}
                          />
                          weeks
                        </div>
                      )}
                    </div>
                  )}
                </div>
              )}
              {results && (
                <div className="sort py-2">
                  <div className="label pr-3 mr-3">
                    <img alt="sort" className={sortBy.order_direction} src={sortIcon}/>
                    <span className="pl-2">Sort</span>
                  </div>
                  <Dropdown
                    options={SORTING_OPTIONS}
                    value={sortBy}
                    valueLabel={({label}) => label}
                    optionLabel={({label}) => label}
                    onChange={sortBy => updateSearchParams({sortBy})}
                  />
                </div>
              )}
              {loading && (
                <div className="loading">
                  <LoadingSpinnerIcon/>
                </div>
              )}
              {results && (
                <ul className="result-list">
                  {results && results.data.map(({price_discount_and_deposit_schedule_hash, ...vehicle}) => {
                    const {
                      subtotal_price_pounds,
                      discount_pounds,
                      driver_price_pounds_after_discount_including_insurance
                    } = price_discount_and_deposit_schedule_hash[isConsumer?number_of_months:number_of_weeks]

                    return (
                      <li key={vehicle.id}>
                        <VehicleCard
                          vehicle={vehicle}
                          isConsumer={isConsumer}
                          totalPrice={Math.ceil(isConsumer?subtotal_price_pounds:driver_price_pounds_after_discount_including_insurance)}
                          discount={discount_pounds}
                        />
                      </li>)
                  })}
                </ul>
              )}
              {results && totalPages && totalPages > 1 && (
                <div className="pagination-wrapper">
                  <p className="number-results">Showing {(page - 1) * per_page + 1}-{page * per_page} of {total_count} results</p>
                  <Pagination
                    activePage={page}
                    itemsCountPerPage={per_page}
                    totalItemsCount={total_count}
                    pageRangeDisplayed={4}
                    onChange={page => updateSearchParams({page})}
                    itemClassPrev="prev"
                    itemClassNext="next"
                  />
                </div>
              )}
            </div>
            <div className="refine-search p-3">
              <Button onClick={this.toggleRefineSearch}>{refineSearch?'Update search and hide filter':'Refine your search'}</Button>
            </div>
          </div>
        </div>
      </SearchFormWrapper>
    )
  }
}

const SearchFormContainer = connect(({vehicle}) => vehicle, {
  updateSearchParams,
  setVehicleType,
  fetchVehicles
})(SearchForm)

export default SearchFormContainer
