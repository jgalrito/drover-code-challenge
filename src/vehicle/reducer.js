import {CLEAR_RESULTS, FETCH_VEHICLES_START, FETCH_VEHICLES_SUCCESS, UPDATE_SEARCH_PARAMS} from './actions'

import {DEFAULT_SEARCH_PARAMS} from './'

export default (state, action) => {
	switch(action.type) {
		case CLEAR_RESULTS: return {
			...state,
			results: null
		};
		case FETCH_VEHICLES_START: return {
			...state,
			loading: true
		};
		case FETCH_VEHICLES_SUCCESS: return {
			...state,
			loading: false,
			results: action.results
		};
		case UPDATE_SEARCH_PARAMS: return {
			...state,
			searchParams: {
				...state.searchParams,
				...action.update
			}
		}
		default: return {
			searchParams: DEFAULT_SEARCH_PARAMS
		}
	}	
}