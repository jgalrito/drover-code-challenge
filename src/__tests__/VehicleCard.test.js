import '../testSetup'

import React from 'react'
import renderer from 'react-test-renderer'

import {VehicleCard} from '../vehicle/VehicleCard'

import results from './mockVehiclesResponse'

it('renders correctly', () => expect(renderer.create(
	<VehicleCard
		vehicle={results.data[0]}
		isConsumer={true}
		totalPrice={50}
		discount={20}
	/>).toJSON()).toMatchSnapshot())