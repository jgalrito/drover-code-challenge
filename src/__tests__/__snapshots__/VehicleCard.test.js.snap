// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="vehicle-card sc-bwzfXH hHSxhc"
>
  <div
    className="slide-show-container"
  >
    <div
      className="vehicle-slide-show sc-htpNat kvSqaQ"
    >
      <div
        className="slick-slider slick-initialized"
        dir="ltr"
      >
        <img
          alt="prev"
          className="slide-arrow prev"
          onClick={[Function]}
          src="left_arrow.png"
        />
        <div
          className="slick-list"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={null}
          onMouseMove={null}
          onMouseUp={[Function]}
          onTouchCancel={null}
          onTouchEnd={[Function]}
          onTouchMove={null}
          onTouchStart={[Function]}
          style={
            Object {
              "padding": "0px 0px",
            }
          }
        >
          <div
            className="slick-track"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            style={
              Object {
                "WebkitTransform": "translate3d(0px, 0px, 0px)",
                "WebkitTransition": "",
                "msTransform": "translateX(0px)",
                "opacity": 1,
                "transform": "translate3d(0px, 0px, 0px)",
                "transition": "",
              }
            }
          >
            <div
              aria-hidden={true}
              className="slick-slide slick-cloned"
              data-index={-2}
              onClick={[Function]}
              style={
                Object {
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <img
                  alt="Exterior - Front"
                  src="https://s3-eu-west-2.amazonaws.com/drover-public/uploads/image/image/20710/main_Screen_Shot_2018-05-24_at_17.47.40.jpg.png"
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                />
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide slick-cloned"
              data-index={-1}
              onClick={[Function]}
              style={
                Object {
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <img
                  alt="Interior - Dashboard"
                  src="https://s3-eu-west-2.amazonaws.com/drover-public/uploads/image/image/20711/main_Screen_Shot_2018-05-24_at_17.47.53.jpg.png"
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                />
              </div>
            </div>
            <div
              aria-hidden={false}
              className="slick-slide slick-active slick-center slick-current"
              data-index={0}
              onClick={[Function]}
              style={
                Object {
                  "outline": "none",
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <img
                  alt="Exterior - Front right side"
                  src="https://s3-eu-west-2.amazonaws.com/drover-public/uploads/image/image/20704/main_Screen_Shot_2018-05-24_at_17.45.49.jpg.png"
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                />
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide"
              data-index={1}
              onClick={[Function]}
              style={
                Object {
                  "outline": "none",
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <img
                  alt="Exterior - Back left side"
                  src="https://s3-eu-west-2.amazonaws.com/drover-public/uploads/image/image/20709/main_Screen_Shot_2018-05-24_at_17.47.03.jpg.png"
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                />
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide"
              data-index={2}
              onClick={[Function]}
              style={
                Object {
                  "outline": "none",
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <img
                  alt="Exterior - Front"
                  src="https://s3-eu-west-2.amazonaws.com/drover-public/uploads/image/image/20710/main_Screen_Shot_2018-05-24_at_17.47.40.jpg.png"
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                />
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide"
              data-index={3}
              onClick={[Function]}
              style={
                Object {
                  "outline": "none",
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <img
                  alt="Interior - Dashboard"
                  src="https://s3-eu-west-2.amazonaws.com/drover-public/uploads/image/image/20711/main_Screen_Shot_2018-05-24_at_17.47.53.jpg.png"
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                />
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide slick-center slick-cloned"
              data-index={4}
              onClick={[Function]}
              style={
                Object {
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <img
                  alt="Exterior - Front right side"
                  src="https://s3-eu-west-2.amazonaws.com/drover-public/uploads/image/image/20704/main_Screen_Shot_2018-05-24_at_17.45.49.jpg.png"
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                />
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide slick-cloned"
              data-index={5}
              onClick={[Function]}
              style={
                Object {
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <img
                  alt="Exterior - Back left side"
                  src="https://s3-eu-west-2.amazonaws.com/drover-public/uploads/image/image/20709/main_Screen_Shot_2018-05-24_at_17.47.03.jpg.png"
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                />
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide slick-cloned"
              data-index={6}
              onClick={[Function]}
              style={
                Object {
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <img
                  alt="Exterior - Front"
                  src="https://s3-eu-west-2.amazonaws.com/drover-public/uploads/image/image/20710/main_Screen_Shot_2018-05-24_at_17.47.40.jpg.png"
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                />
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide slick-cloned"
              data-index={7}
              onClick={[Function]}
              style={
                Object {
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <img
                  alt="Interior - Dashboard"
                  src="https://s3-eu-west-2.amazonaws.com/drover-public/uploads/image/image/20711/main_Screen_Shot_2018-05-24_at_17.47.53.jpg.png"
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                />
              </div>
            </div>
          </div>
        </div>
        <img
          alt="next"
          className="slide-arrow next"
          onClick={[Function]}
          src="left_arrow.png"
        />
      </div>
    </div>
    <img
      alt="discount"
      src="discount_tag.png"
    />
  </div>
  <div
    className="vehicle-info p-1 p-lg-0"
  >
    <div
      className="vehicle-title pl-3"
    >
      <div
        className="pb-1 pt-2"
      >
        <h3
          className="vehicle-make-brand m-0"
        >
          Toyota Aygo X-Play 1.0L
        </h3>
        <span
          className="vehicle-location"
        >
          Located in 
          E14
        </span>
      </div>
      <div
        className="availability pr-3"
      >
        Available from 
        8th June 2018
      </div>
    </div>
    <div
      className="key-facts py-1 px-2"
    >
      <ul>
        <li
          className="px-1 pb-1"
        >
          <img
            alt="2014"
            src="year.svg"
          />
          <span>
            2014
          </span>
        </li>
        <li
          className="px-1 pb-1"
        >
          <img
            alt="Hatchback"
            src="body_type.svg"
          />
          <span>
            Hatchback
          </span>
        </li>
        <li
          className="px-1 pb-1"
        >
          <img
            alt="Manual"
            src="transmission.svg"
          />
          <span>
            Manual
          </span>
        </li>
        <li
          className="px-1 pb-1"
        >
          <img
            alt="Petrol"
            src="fuel_type.svg"
          />
          <span>
            Petrol
          </span>
        </li>
        <li
          className="px-1 pb-1"
        >
          <img
            alt="4 Seats"
            src="seats.svg"
          />
          <span>
            4 Seats
          </span>
        </li>
        <li
          className="px-1 pb-1"
        >
          <img
            alt="5 Doors"
            src="doors.svg"
          />
          <span>
            5 Doors
          </span>
        </li>
        <li
          className="px-1 pb-1"
        >
          <img
            alt="Silver"
            src="color.svg"
          />
          <span>
            Silver
          </span>
        </li>
      </ul>
    </div>
    <ul
      className="features py-1 px-2"
    >
      <li
        className="pl-1 pr-1"
      >
        Bluetooth Connectivity
      </li>
      <li
        className="pl-1 pr-1"
      >
        •
      </li>
      <li
        className="pl-1 pr-1"
      >
        Air Conditioning
      </li>
      <li
        className="pl-1 pr-1"
      >
        •
      </li>
      <li
        className="pl-1 pr-1"
      >
        Upholstery Cloth
      </li>
      <li
        className="pl-1 pr-1"
      >
        •
      </li>
      <li
        className="pl-1 pr-1"
      >
        FM/AM Radio
      </li>
      <li
        className="pl-1 pr-1"
      >
        •
      </li>
      <li
        className="pl-1 pr-1"
      >
        Aux Input
      </li>
    </ul>
    <div
      className="pricing px-2 pb-2 pt-0 p-lg-2 p-xl-2 px-xl-3"
    >
      <div
        className="discount-info"
      >
        <div>
          <small
            className="max-price"
          >
            was £
            70
            /month
          </small>
        </div>
        <div>
          <small>
            (save £
            20
            /month)
          </small>
        </div>
      </div>
      <div
        className="value-wrapper mr-lg-3 p-2 p-lg-0 mb-2 mb-lg-0"
      >
        <small
          className="max-price"
        >
          £
          70
        </small>
        <div
          className="value"
        >
          <span
            className="price text-xl-right"
          >
            £ 
            50
          </span>
          /month
        </div>
        <small
          className="periodicity"
        >
          (Monthly Vehicle Price inc. VAT)
        </small>
      </div>
      <button
        className="sc-bdVaJa hRfSQH"
      >
        See more details
      </button>
    </div>
  </div>
</div>
`;
